client/ayon_airtable/plugins/publish/collect_product_name_for_editorial_return.py

```diff
@@ -0,0 +1,71 @@
+
+
+"""Plugin to collect product names for editorial return from Airtable.
+
+This module defines a Pyblish context plugin that retrieves product names
+from Airtable records based on the current project and exposes them for
+user selection during the publishing process.
+"""
+
+import os
+
+import pyblish.api
+from ayon_airtable.common.airtable_api_handlers import get_airtable_table
+from ayon_core.lib import EnumDef
+from ayon_core.pipeline import get_current_project_name
+
+
+class CollectProductNameForEditorialReturn(pyblish.api.ContextPlugin):
+    """Collects product name from users for editorial return."""
+    order = pyblish.api.CollectorOrder
+    label = "Collect Product Name For Editorial Return (Airtable)"
+    families = ["editorial", "editorial_pkg"]  # noqa: RUF012
+
+    def process(self, context: pyblish.api.Context) -> None:
+        """Collect product names from the user's preference.
+
+        Args:
+            context (pyblish.api.Context): The context of the plugin.
+        """
+        airtable_table = get_airtable_table()
```

Comment:
This call relies on environment variable set. I would be better to collect everything needed into context/instance and then pass it to the call.